context/app/static/js/hooks/useCopyText.ts

```diff
@@ -6,17 +6,19 @@ export const useHandleCopyClick = () => {
   const { toastSuccess, toastError } = useSnackbarStore();
   return useCallback(
     (value: string, additionalText?: string) => {
-      navigator.clipboard
-        .writeText(value)
-        .then(() => {
-          toastSuccess(`Copied to clipboard.${additionalText ? ` ${additionalText}` : ''}`);
-        })
-        .catch((e) => {
-          console.error('Error copying to clipboard', e);
-          toastError(
-            `Encountered an error while copying, please try again.${additionalText ? ` ${additionalText}` : ''}`,
-          );
-        });
+      setTimeout(() => {
```

Comment:
We should document the necessity of this `setTimeout`, e.g. how it fixes the issue at hand.

Reply:
Agreed - not entirely sure why this fix works, I followed [this thread](https://stackoverflow.com/questions/62327358/javascript-clipboard-api-safari-ios-notallowederror-message) about the console error that was provided. Looks like wrapping the call in a `setTimeout` re-contextualizes this action as a user event?